packages/components/src/button/index.native.js

```diff
@@ -93,9 +93,9 @@ export function Button( props ) {
 
 	const subscriptInactive = getStylesFromColorScheme( styles.subscriptInactive, styles.subscriptInactiveDark );
 
-	const newChildren = Children.map( compact( children ), ( child ) => {
+	const newChildren = Array.isArray( children ) ? Children.map( compact( children ), ( child ) => {
```

Comment:
`Children.map` should already be doing the right thing here, I think the problem is that instead of passing `children` as is, we are passing `compact( children )`, which is the part assuming that `children` is an array.

Have you tried removing that `compact` from the previous code?

Reply:
We can not remove `compact` since there is a possibility that the children array contains `undefined` or `null'` which crashes with `cloneElement` function. What I can do here is:

```js
const newChildren = Children.map( children, ( child ) => {
  return child ?
    cloneElement( child, { colorScheme: props.preferredColorScheme, active: ariaPressed } ) :
    null;
} )
```
WDYT?

Reply:
I'm not sure in which cases would that happen, but I agree that it's not impossible. I think your new solution looks simpler than the previous one 👍 

Instead of `null`, I think we should return `child` in case it's `undefined` and the distinction matters to something else.

Reply:
Changed :) 
w/o `compact` app crashed right after the start so it is quite common. `undefined` or `null` exists in `children` when you do something like this in render:
```js
<Button>
  { something ? <Icon /> : null }
  <Text>click</Text> 
</Button>
```
```js
<Button>
  { something && <Icon /> }
  <Text>click</Text> 
</Button>
```
And value of `something` is `null` or `undefined`